pkg/config/mapper.go

```diff
@@ -0,0 +1,98 @@
+package config
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/logandavies181/arnlike"
+)
+
+// SSOArnLike returns a string that can be passed to arnlike.ArnLike to
+// match canonicalized IAM Role ARNs against. Assumes Validate() has been called.
+func (m *RoleMapping) SSOArnLike() string {
+	if m.SSO == nil {
+		return ""
+	}
+
+	var partition string
+	if m.SSO.Partition == "" {
+		partition = "aws"
+	}
+
+	return strings.ToLower(fmt.Sprintf("arn:%s:iam::%s:role/AWSReservedSSO_%s_*", partition, m.SSO.AccountID, m.SSO.PermissionSetName))
+}
+
+// Validate returns an error if the RoleMapping is not valid after being unmarshaled
+func (m *RoleMapping) Validate() error {
+	if m == nil {
+		return fmt.Errorf("RoleMapping is nil")
+	}
+
+	if m.RoleARN == "" && m.SSO == nil {
+		return fmt.Errorf("One of rolearn or SSO must be supplied")
+	} else if m.RoleARN != "" && m.SSO != nil {
+		return fmt.Errorf("Only one of rolearn or SSO can be supplied")
+	}
+
+	if m.SSO != nil {
+		ssoArnLikeString := m.SSOArnLike()
+		ok, err := arnlike.ArnLike(ssoArnLikeString, "arn:*:iam:*:*:role/*")
```

Comment:
To continue my comment from above, I *think* that if a k8s admin set `sso.accountID` or `sso.permissionSetName` to `*`, this `arnlike.ArnLike()` check would return `true, nil`, right? Wouldn't that be a way around the whole "make sure an account ID and a permission set name is specified" thing?

Reply:
This is more a sanity check that the settings look like a role when parsed. I've added some checks for those settings in Validate() now